### PR TITLE
chore: update to Rust 1.58 and switch to GCP Rust crate

### PR DESCRIPTION
**Pull request by ethowitz**
## Description
Update to Rust 1.58 and start using the google-cloud-rust-raw crate instead of vendoring the GCP code.

## Testing

N/A

## Issue(s)

N/A

### CODE REVIEW

**Comment by jrconlin** [comment]
oof. Looks like a minor `fmt` fail. 
Also, did you want to bump things up to `edition="2021"` while you have the chance? From what I've seen, there's no impact to existing code from `fmt` or `clippy`

**Comment by jrconlin** [comment]
Sigh, and looks like after a `cargo update`, there's a new version of `protobuf`.

**Comment by ethowitz** [comment]
I suppose we'll need to rebuild the GCP Rust bindings then?

**Comment by jrconlin** [comment]
Not necessarily, or at least, not a full rebuild should be required. There are steps in the vendored directory you can do. 
 I can also try doing a PR that is just that part.

**Comment by ethowitz** [comment]
I'll give it a shot!

**Review by pjenvey — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM rust:1.57-buster as builder
+FROM rust:1.58-buster as builder
```
[the remark this answers is not in the document]
Comment (jrconlin) [reply Dockerfile:1]: I think that if you specify 1.58, it pulls in the latest version subversion.

[the remark this answers is not in the document]
Comment (jrconlin) [reply Dockerfile:1]: Just did a `docker pull rust:1.58-buster` and an internal shell 
```rustup show``` shows 1.58.1

**Review by pjenvey — APPROVED** [review]

**Pull request merged**